api/src/client.rs

```diff
@@ -352,6 +354,26 @@ impl ApiClient for Client {
         }
         Ok(res.diff)
     }
+
+    async fn validate_module(
+        &self,
+        wasm: impl AsRef<[u8]> + Send,
+        checkfile: impl AsRef<[u8]> + Send,
+    ) -> Result<Report> {
+        let input = api::validate_module_request::Module_input::Module(wasm.as_ref().to_vec());
+        let req = api::ValidateModuleRequest {
+            checkfile: checkfile.as_ref().to_vec(),
+            module_input: Some(input),
+            ..Default::default()
+        };
+
+        let res: api::ValidateModuleResponse =
+            self.send(ModserverCommand::ValidateModule(req)).await?;
+        if res.error.is_some() {
+            return Err(api_error(res.error, "validate request failed"));
+        }
+        Ok(serde_json::from_slice(&res.invalid_module_report)?)
```

Comment:
This tripped me up because I thought `invalid_module_report` meant that the validation failed, but I see now that it can just be empty, indicating no validation errors.  Maybe `module_report`  would be a better name but not as part of this PR